pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -276,13 +276,34 @@ func (p *PriorityQueue) isPodBackingOff(pod *v1.Pod) bool {
 
 // backoffPod checks if pod is currently undergoing backoff. If it is not it updates the backoff
 // timeout otherwise it does nothing.
-func (p *PriorityQueue) backoffPod(pod *v1.Pod) {
-	p.podBackoff.CleanupPodsCompletesBackingoff()
+func (p *PriorityQueue) backoffPod(podInfo *framework.PodInfo) {
+	p.cleanupPodsCompletesBackingoff()
 
-	podID := nsNameForPod(pod)
+	podID := nsNameForPod(podInfo.Pod)
 	boTime, found := p.podBackoff.GetBackoffTime(podID)
 	if !found || boTime.Before(p.clock.Now()) {
-		p.podBackoff.BackoffPod(podID)
+		p.podBackoff.BackoffPod(podID, podInfo)
+	}
+}
+
+// cleanupPodsCompletesBackingoff executes garbage collection on PodBackoffMap,
+// as well as removes pod from either podBackoffQ or unschedulableQ
+func (p *PriorityQueue) cleanupPodsCompletesBackingoff() {
+	p.podBackoff.lock.Lock()
+	defer p.podBackoff.lock.Unlock()
+
+	for nsNameForPod, podInfo := range p.podBackoff.podInfoMap {
```

Comment:
Aren't we losing the attempts information when doing this?

Basically, any time we hit the following timeouts: 1s, 2s, 4s, 8s, we go back to 1s again.

On the other hand, it looks like PodInfo has attempts information already. We should reuse that instead of the map we keep in PodBackoffMap.

In fact, we should add a unit test that exercises this.

Reply:
I'm adding a unit test for this in #87616